CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKCanvas.java

```diff
@@ -5,83 +5,115 @@
  */
 package au.gov.asd.tac.constellation.visual.vulkan;
 
+import au.gov.asd.tac.constellation.visual.vulkan.renderables.CVKRenderable;
+import au.gov.asd.tac.constellation.visual.vulkan.utils.CVKGraphLogger;
+import static au.gov.asd.tac.constellation.visual.vulkan.utils.CVKUtils.CVKAssertNotNull;
 import java.awt.Graphics;
+
 import org.lwjgl.vulkan.awt.AWTVKCanvas;
 import javax.swing.SwingUtilities;
+import static org.lwjgl.vulkan.VK10.VK_NULL_HANDLE;
 import org.lwjgl.vulkan.awt.VKData;
 
 
-
-public class CVKCanvas extends AWTVKCanvas{
-    private boolean parentAdded = false;
-    private int frameNumber = 0;
+public class CVKCanvas extends AWTVKCanvas {
+    private final CVKVisualProcessor cvkVisualProcessor;    
```

Comment:
default to null?

Reply:
No it's final, think of it like a reference, it must be assigned to in the ctor or else it won't compile.